File: nhglib-core/src/io/github/movementspeed/nhglib/runtime/ecs/interfaces/RenderingSystemInterface.java
package io.github.movementspeed.nhglib.runtime.ecs.interfaces;

import com.badlogic.gdx.graphics.g3d.RenderableProvider;
import com.badlogic.gdx.utils.Array;

public interface RenderingSystemInterface {
    void onPreRender();

    void onRender();

    void onPostRender();
    void clearRenderableProviders();

    void onUpdatedRenderer(int renderingWidth, int renderingHeight);
    Array<RenderableProvider> getRenderableProviders();
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/ecs/interfaces/RenderingSystemInterface.kt
package io.github.movementspeed.nhglib.core.ecs.interfaces

import com.badlogic.gdx.graphics.g3d.RenderableProvider
import com.badlogic.gdx.utils.Array

interface RenderingSystemInterface {

    val renderableProviders: Array<RenderableProvider>
    fun onPreRender()

    fun onPostRender()

    fun clearRenderableProviders()

    fun onUpdatedRenderer(renderingWidth: Int, renderingHeight: Int)
}